karapace/auth.py

```diff
@@ -87,24 +88,31 @@ async def start_refresh_task(self, stats: StatsClient) -> None:
 
         async def _refresh_authfile() -> None:
             """Reload authfile, but keep old auth data if loading fails"""
-            try:
-                # Poll delay for fallback mechanism if system notifications cannot be used.
-                async for changes in awatch(
-                    self._auth_filename, poll_delay_ms=5000, stop_event=self._refresh_auth_awatch_stop_event
-                ):
-                    unique_files = {filename for _, filename in changes}
-                    for filename in unique_files:
-                        if self._auth_filename in filename:
-                            try:
-                                self._load_authfile()
-                            except InvalidConfiguration as e:
-                                log.warning("Could not load authentication file: %s", e)
-            except asyncio.CancelledError:
-                log.info("Closing schema registry ACL refresh task")
-                return
-            except Exception as ex:  # pylint: disable=broad-except
-                log.exception("Schema registry auth file could not be loaded")
-                stats.unexpected_exception(ex=ex, where="schema_registry_authfile_reloader")
+            while True:
+                try:
+                    async for changes in awatch(self._auth_filename, stop_event=self._refresh_auth_awatch_stop_event):
+                        changes_by_filename = defaultdict(set)
+                        for change, filename in changes:
+                            changes_by_filename[filename].add(change)
```

Comment:
It looks a bit odd to me to be dealing with multiple filenames here, when we're only ever interested in a single one.

Perhaps instead we could just flatten the data into a set for only the interesting file?

```python
changes = {
    change
    for  change, filename in changes
    if self._auth_filename in filename
}
if not changes:
    continue
```

I think we'd then be able to skip the for loop and conditional guarding `self._load_authfile()` too?

This would also allow simplifying the `raise StopIteration` into just `break`, I think?